### ISSUE TITLE
[Bug] convert mmseg checkpoint to TRT error, when setting use_sigmoid=True

### ISSUE THREAD

**Issue by vincentdeeplearn** [open-mmlab/mmdeploy#2353]
### Checklist

- [X] I have searched related issues but cannot get the expected help.
- [X] 2. I have read the [FAQ documentation](https://github.com/open-mmlab/mmdeploy/tree/main/docs/en/faq.md) but cannot get the expected help.
- [X] 3. The bug has not been fixed in the latest version.

### Describe the bug

1. I set use  sigmoid=True to train a model by mmsegmention repo.
2. then  i use mmploy to convert the *.pth to TRT engine(sucess, no error), but infer using tensorRT, the result is wrong.
3. I can get output_pytorch.jpg infered right, but output_tensorrt.jpg is error. 
4. debug with netron to review the *.onnx, find the final output node activate function is softmax, not sigmoid.

### Reproduction

1. set the mmseg config and trained  a model using custom datasets.
- decode_head=dict(
        num_classes=2,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0),
        out_channels=1)

2. using mmdeploy convet *.pth to TRT engine.

python deploy.py ~/mmdeploy/configs/mmseg/segmentation_tensorrt-fp16_static-640x1408.py   deeplabv3plus_r50-d8_512x1024_80k_cityscapes_trt.py  iter_50000.pth  ~/test_img.jpg  --work-dir  ./trt_engine_fp16_640x1408  --device cuda --dump-info

#### But when we close use_sigmoid: *we can infer right by trt engine!*
- decode_head=dict(
        num_classes=2,
        loss_decode=dict(
        type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0)),

### Environment

```Shell
2023-08-14 10:50:17,021 - mmdeploy - INFO - **********Environmental information**********
fatal: not a git repository (or any of the parent directories): .git
2023-08-14 10:50:17,511 - mmdeploy - INFO - sys.platform: linux
2023-08-14 10:50:17,512 - mmdeploy - INFO - Python: 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 06:04:10) [GCC 10.3.0]
2023-08-14 10:50:17,512 - mmdeploy - INFO - CUDA available: True
2023-08-14 10:50:17,512 - mmdeploy - INFO - GPU 0: NVIDIA GeForce RTX 3090
2023-08-14 10:50:17,512 - mmdeploy - INFO - CUDA_HOME: /usr/local/cuda
2023-08-14 10:50:17,512 - mmdeploy - INFO - NVCC: Cuda compilation tools, release 11.6, V11.6.124
2023-08-14 10:50:17,512 - mmdeploy - INFO - GCC: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
2023-08-14 10:50:17,512 - mmdeploy - INFO - PyTorch: 1.13.1+cu116
2023-08-14 10:50:17,512 - mmdeploy - INFO - PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.6
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.3.2  (built against CUDA 11.5)
  - Magma 2.6.1
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.6, CUDNN_VERSION=8.3.2, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -fabi-version=11 -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wunused-local-typedefs -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.13.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

2023-08-14 10:50:17,512 - mmdeploy - INFO - TorchVision: 0.14.1+cu116
2023-08-14 10:50:17,512 - mmdeploy - INFO - OpenCV: 4.2.0
2023-08-14 10:50:17,512 - mmdeploy - INFO - MMCV: 1.7.1
2023-08-14 10:50:17,513 - mmdeploy - INFO - MMCV Compiler: GCC 9.3
2023-08-14 10:50:17,513 - mmdeploy - INFO - MMCV CUDA Compiler: 11.6
2023-08-14 10:50:17,513 - mmdeploy - INFO - MMDeploy: 0.14.0+
2023-08-14 10:50:17,513 - mmdeploy - INFO - 

2023-08-14 10:50:17,513 - mmdeploy - INFO - **********Backend information**********
2023-08-14 10:50:17,594 - mmdeploy - INFO - tensorrt:   8.2.4.2
2023-08-14 10:50:17,594 - mmdeploy - INFO - tensorrt custom ops:        Available
2023-08-14 10:50:17,673 - mmdeploy - INFO - ONNXRuntime:        1.12.1
2023-08-14 10:50:17,673 - mmdeploy - INFO - ONNXRuntime-gpu:    None
2023-08-14 10:50:17,673 - mmdeploy - INFO - ONNXRuntime custom ops:     NotAvailable
2023-08-14 10:50:17,683 - mmdeploy - INFO - pplnn:      None
2023-08-14 10:50:17,690 - mmdeploy - INFO - ncnn:       None
2023-08-14 10:50:17,692 - mmdeploy - INFO - snpe:       None
2023-08-14 10:50:17,694 - mmdeploy - INFO - openvino:   None
2023-08-14 10:50:17,695 - mmdeploy - INFO - torchscript:        1.13.1+cu116
2023-08-14 10:50:17,696 - mmdeploy - INFO - torchscript custom ops:     NotAvailable
2023-08-14 10:50:17,785 - mmdeploy - INFO - rknn-toolkit:       None
2023-08-14 10:50:17,785 - mmdeploy - INFO - rknn2-toolkit:      None
2023-08-14 10:50:17,797 - mmdeploy - INFO - ascend:     None
2023-08-14 10:50:17,798 - mmdeploy - INFO - coreml:     None
2023-08-14 10:50:17,799 - mmdeploy - INFO - tvm:        None
2023-08-14 10:50:17,799 - mmdeploy - INFO - 

2023-08-14 10:50:17,799 - mmdeploy - INFO - **********Codebase information**********
2023-08-14 10:50:17,802 - mmdeploy - INFO - mmdet:      2.28.2
2023-08-14 10:50:17,802 - mmdeploy - INFO - mmseg:      0.30.0
2023-08-14 10:50:17,802 - mmdeploy - INFO - mmcls:      None
2023-08-14 10:50:17,802 - mmdeploy - INFO - mmocr:      None
2023-08-14 10:50:17,802 - mmdeploy - INFO - mmedit:     None
2023-08-14 10:50:17,802 - mmdeploy - INFO - mmdet3d:    None
2023-08-14 10:50:17,802 - mmdeploy - INFO - mmpose:     None
2023-08-14 10:50:17,802 - mmdeploy - INFO - mmrotate:   None
2023-08-14 10:50:17,802 - mmdeploy - INFO - mmaction:   None
```

### Error traceback

load checkpoint from local path: /workspace/data/networks/mmsegmentation/adapter_tag/deeplabv3plus_2023_08_11_03_09/iter_50000.pth
2023-08-14 10:47:32,478 - mmdeploy - INFO - visualize pytorch model success.
2023-08-14 10:47:32,479 - mmdeploy - INFO - All process success.
_No response_

**Comment by RunningLeon** [comment]
@vincentdeeplearn hi, in case you are using master branch, you could git pull the code after this PR https://github.com/open-mmlab/mmdeploy/pull/1997

**Comment by vincentdeeplearn** [comment]
> @vincentdeeplearn hi, in case you are using master branch, you could git pull the code after this PR #1997

ok!  Thanks. Let me try.

**Comment by vincentdeeplearn** [comment]
Done! thanks.
